temporal-sdk/src/test/java/io/temporal/workflow/nexus/TerminateWorkflowAsyncOperationTest.java

```diff
@@ -54,10 +55,19 @@ public void terminateAsyncOperation() {
         Assert.assertThrows(WorkflowFailedException.class, () -> workflowStub.execute(""));
     Assert.assertTrue(exception.getCause() instanceof NexusOperationFailure);
     NexusOperationFailure nexusFailure = (NexusOperationFailure) exception.getCause();
-    Assert.assertTrue(nexusFailure.getCause() instanceof ApplicationFailure);
-    Assert.assertEquals(
-        "operation terminated",
-        ((ApplicationFailure) nexusFailure.getCause()).getOriginalMessage());
+    // TODO: Test server needs to be fixed to return the correct type
```

Comment:
If this TODO is to be done separately, can we open an issue and reference it here?